File: 2022/Day1/python/Day1.py
# Part 1 and Part 2
# Zheng Lin Lei


from pathlib import Path
import time

SCRIPT_DIR = Path(__file__).parent
INPUT_FILE = Path(SCRIPT_DIR, "../input.txt")

def main():
    with open(INPUT_FILE, mode="rt") as f:
        elf_meals = f.read().split("\n\n") # split on empty lines
    
    elf_calories = [] # store total calories for each elf
    for elf in elf_meals:
        calories = sum(map(int, elf.splitlines()))
        elf_calories.append(calories)
        
    print(f"Part 1: {max(elf_calories)}")
    
    elf_calories = sorted(elf_calories)
    print(f"Part 2: {elf_calories[-3:]}")
    print(f"Part 2: {sum(elf_calories[-3:])}")
    

if __name__ == "__main__":
    t1 = time.perf_counter()
    main()
    t2 = time.perf_counter()
    print(f"Execution time: {t2 - t1:0.4f} seconds")

File: 2022/Day1/javascript/Day1.js
// Read File ../input.txt

// Part 1 and Part 2


// ZhengLinLei

const fs = require('fs');

// Read File Async
fs.readFile('../input.txt', 'utf8', (err, data) => {
    if (err) {
        console.error(err)
        return
    }
    let data_filtered = data.split('\n');

    let sum_data = [];
    let sum = 0;

    // Foreach each item and sum it
    for(let i = 0; i < data_filtered.length; i++) {
        if(data_filtered[i] == ""){
            sum_data.push(sum);
            sum = 0;
        }else{
            sum += parseInt(data_filtered[i]);
        }
    }




    console.log("Part 1: ", Math.max(...sum_data));

    let sorted_d = sum_data.sort().reverse();
    // Sum top 3
    console.log("Part 2: " + (sorted_d[0] + sorted_d[1] + sorted_d[2]));
});



